xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -4440,7 +4440,11 @@ bool CVideoPlayer::OnAction(const CAction &action)
     }
   }
 
-  if (dynamic_cast<CDVDInputStreamPVRManager*>(m_pInputStream))
+  //If the current channel is a nullptr, and this is a PVRStream, then this is a saved recording
+  CDVDInputStreamPVRManager* pvrStream = dynamic_cast<CDVDInputStreamPVRManager*>(m_pInputStream);
+  CPVRChannelPtr channel = g_PVRManager.GetCurrentChannel();
+  // If the pvrStream is not a saved recording, then handle the actions below
+  if (pvrStream && channel != nullptr)
```

Comment:
in general shared pointers are tested for being empty with a boolean operator. comparing with nullptr is not necessarily the same.
if (pvrStream && channel)
{